src/main/java/com/redislabs/Connector.java

```diff
@@ -289,11 +287,19 @@ public void foreach(ArrayList<HashMap<String, Object>> record) throws Exception
         transaction.commit();
         session.clear();
       }catch (Exception e) {
-        msg = String.format("Failed commiting transaction error='%s'", e.toString());
+        msg = String.format("Failed commiting transaction error='%s'", e);
         GearsBuilder.log(msg, LogLevel.WARNING);
-        connector.closeSession();
+
+        //connector.closeSession();
+        if ( connector.getSession().getTransaction().isActive() ) {
+          connector.getSession().getTransaction().rollback();
+        }
+
+        connector.getSession().clear();
         lastStreamId = null;
         cause = e;
+
+        retry(record.subList(lastCommittedIdx + 1, record.size()));
```

Comment:
We should do the retry only if it was requested when register the connector.

Reply:
Added an option to use DLQ

Reply:
I do not see where you use this new value?

Reply:
Right, I moved it to separate exceptions and forgot to put it back in. Just did